topics/go/algorithms/data/hash/hash.go

```diff
@@ -0,0 +1,107 @@
+// Package hash implements a hash table
+package hash
+
+import (
+	"fmt"
+	"hash/maphash"
+)
+
+const (
+	numBuckets = 256 // Number of buckets in hash
+)
+
+// entry in the hash table
+type entry struct {
+	key   string
+	value int
+}
+
+// Hash is a simple Hash table
+type Hash struct {
+	buckets [][]*entry
+	h       maphash.Hash
+}
+
+// NewHash returns a new hash table
+func NewHash() *Hash {
+	return &Hash{
+		buckets: make([][]*entry, numBuckets),
+	}
+}
+
+// Set sets the key to value in the hash
+func (h *Hash) Set(key string, value int) {
+	b := h.hashKey(key)
+	// Look for existing key
+	for _, e := range h.buckets[b] {
```

Comment:
Wondering if we don't use `entry` over `e` for clarity. This is complicated stuff.

Actually. If there is a comment that says this the variables become self evident.

// Iterate over the entries for the specified bucket.

Reply:
OK